### ISSUE TITLE
contract state is wrong after a transfer

### ISSUE THREAD

**Issue by nicbus** [RGB-WG/rgb#187]
Using the `develop` branch, commit `4fc24b1`, I can complete a transfer but the final balances are wrong (0 on both sides).

This is the contract state of the sender after the transfer:
```
Loading descriptor from wallet wallet_0 ... success                                                                                                                                                                                                                                                                            
Global:                                                                                                                                                                                                                                                                                                                        
  spec := (ticker=("USDT"), name=("USD Tether"), details=~, precision=0)                                                                                                                                                                                                                                                       
  terms := (text=("demo NIA asset"), media=~)                                                                                                                                                                                                                                                                                  
  issuedSupply := (2000)                                                                                                                                                                                                                                                                                                       
                                                                                                                                                                                                                                                                                                                               
Owned:                                                                                                                                                                                                                                                                                                                         
  assetOwner:                                                                                                                                                                                                                                                                                                                  
    amount=2000, utxo=bc:opret1st:ce620aa0b5ed49a36d8d72951d4c9242a835ea8fdedd067f5cbdcf84086036c1:1, witness=~ # owner unknown                                                                                                                                                                                                
```

This is the contract state of the recipient after the transfer:
```
Loading descriptor from wallet wallet_1 ... success                                                                                                                                                                                                                                                                            
Global:                                                                                                                                                                                                                                                                                                                        
  spec := (ticker=("USDT"), name=("USD Tether"), details=~, precision=0)                                                                                                                                                                                                                                                       
  terms := (text=("demo NIA asset"), media=~)                                                                                                                                                                                                                                                                                  
  issuedSupply := (2000)                                                                                                                                                                                                                                                                                                       
                                                                                                                                                                                                                                                                                                                               
Owned:                                                                                                                                                                                                                                                                                                                         
  assetOwner:                                                                                                                                                                                                                                                                                                                  
    amount=1900, utxo=bc:opret1st:3cb6ddaad92b1e5536f7d9a7b1ccd00bf49583c1eb3385a705220329b247619a:0, witness=bc:3cb6ddaad92b1e5536f7d9a7b1ccd00bf49583c1eb3385a705220329b247619a # owner unknown                                                                                                                              
    amount=2000, utxo=bc:opret1st:ce620aa0b5ed49a36d8d72951d4c9242a835ea8fdedd067f5cbdcf84086036c1:1, witness=~ # owner unknown                                                                                                                                                                                                
```

Note: I have use the `-a` option to show all state.

**Comment by nicbus** [comment]
Re-testing with the fix in `8b7506f` (plus a cargo update of `bp-wallet` to include fixes up to `03bb754`) the `transfer` command now fails with:
```
Loading descriptor from wallet wallet_0 ... success
thread 'main' panicked at ../rgb/src/store.rs:241:20:
error saving data: "unable to serialize wallet data as TOML file - map key was not a string"
```

**Comment by nicbus** [comment]
Re-tested using commit `d6c4228` and the issue is now fixed.
